### DISCUSSION TITLE
Spezi Modules and Components for HealthKit

### DISCUSSION THREAD

**Discussion by Shaikmasaqhlain888** [StanfordSpezi/.github#61] [Challenges and Help] [answered]
### Description

While I integrated the SPEZI Framework in the my project, I wanna know all the modules that spezi provides for specific functionalities eg: using Spezi "some Module" instead of HealthKit and What components I can get from Spezi for "watch connectivity". This makes me wonder if there is any file which lists  all the functionalities and syntax of the components in Spezi.

### Contact Information and Additional Context

This is the very beginning of the project hence this discussion would be very useful in near future for all my challenges.

### Code of Conduct

- [X] I agree to follow this projects's Code of Conduct and Contributing Guidelines

**Comment by PSchmiedmayer** [comment, marked as answer]
Hi @Shaikmasaqhlain888; thank you for reaching out!

You can find an overview of all our iOS modules on the [Swift Package Index](https://swiftpackageindex.com/StanfordSpezi). All the documentation for the Swift Packages helps you to find out more about the package, e.g., the Spezi HealthKit module: https://swiftpackageindex.com/StanfordSpezi/SpeziHealthKit/documentation/spezihealthkit.

The goal with Spezi is to provide building blocks for your software system, and we aim to provide elements where we can provide value and simplify setups. So far, we have not identified an element where we can simply and improve Apple Watch connectivity; I would suggest using the Apple Watch Connectivity framework: https://developer.apple.com/documentation/watchconnectivity.
That's one of the great strengths of our modular setup; you still have a Swift/Kotlin Application as your foundation, and you can implement everything you need beyond Spezi modules as you would for every other mobile application.

Our [Spezi](https://github.com/StanfordSpezi/Spezi) repository as well as our [Spezi Template Application](https://github.com/StanfordSpezi/SpeziTemplateApplication) are great starting points to dive into most modules and have a starting point to explore functionality.
